mitodl/bootcamp-ecommerce
# Cleaned up tos / privacy policy link usage

#### What are the relevant tickets?
Fixes #783 

#### What's this PR do?

This PR covers a fair amount of cleanup around TOS and related links:
- Add a custom setting page (`SitePageSettings`) to support configuring which pages are TOS, Privacy Policy, etc
- Rolled `main.views.get_base_context` into `main.context_processors` so that these get automatically added to all views (we were manually doing this in all views before).
- Cleaned up the email verification template, which had a duplicate (but more correct) footer in it

#### How should this be manually tested?

- Login to the CMS, there should be a "Resource Pages" page under "Settings"
  - For each of the options listed, create a page of the `ResourcePage` type and associate it on this page
  - Save the new settings
- Poke about the app, including logging into `/applications` and verify the context processor changes didn't break any existing views.
- Verify the links in the footer go where you'd expect them to
- Go to `/__emaildebugger__/` and verify for each of the emails the footer links render correctly
  - Also verify the fixes for the verification template noted above
- Create a Letter template page and verify the preview link still works

#### Any background context you want to provide?

A simpler approach to this would've been to just update the current hardcoded links, I opted for the custom setting approach as that seemed a) more flexible, leaving the url/slug up to CMS users and b) more idiomatic to how Wagtail likes to set things up. It meant a bit of extra work, but it means we'll never have to update these urls in the future if a different path is desired. It also sets a best-practice for how we represent these in the app.
